src/utils.ts

```diff
@@ -173,6 +195,37 @@ export async function resolveLabel(label: string, config: Config): Promise<strin
   return config.provider.resolveName(`${label}.${config.registrar.domain}`);
 }
 
+export async function lookupAddress(address: string, config: Config): Promise<Profile | null>  {
+  const profile: Profile = {} as Profile;
+  try {
+
+    const ens = await resolveEnsProfile(address, config);
+    if (ens) profile.ens = ens;
+    const idx = await resolveIdxProfile(formatCAIP10Address(address, "eip155", 3), config);
+    if (idx) profile.idx = idx;
+  } catch (error) {
+    console.error(error);
+  }
+
+  console.log("Profiles:", profile);
+
+  return profile || null;
```

Comment:
Not sure what you expect here, but isn't `{} || null` going to return `{}`?

I think we should just return `Profile` from this function anyway. The ui code can check if the ens or idx keys are set.

Reply:
Yeah you're right, wanted to be explicit first but, even if nothing gets resolved it returns an empty object which is not null